src/main/java/nwoolcan/model/brewery/BreweryImpl.java

```diff
@@ -70,4 +76,17 @@ public Warehouse getWarehouse() {
     public void addBatch(final Batch newBatch) {
         batches.add(newBatch);
     }
+
+    @Override
+    public Result<Empty> stockBatch(final Batch batch, final BeerArticle beerArticle, @Nullable final Date expirationDate) {
+        return Result.of(batch)
+                     .require(batch::isEnded)
```

Comment:
No need to check for this, but not wrong to do

Reply:
whyNot?